### PR TITLE
txDAG transfer by gasless transaction (#28)

### PR DESCRIPTION
**Pull request by andyzhang2023**
* txDAG transfer

* encode/decode txDAG data with ABI

* set txDAG receiver to a special address

---------

Co-authored-by: andyzhang2023 <andyzhang2023@gmail.com>

merge conflict

fix conflict

### Description

add a description of your changes here...

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes:
* add each change in a bullet point here
* ...

### CODE REVIEW

File: miner/worker.go [hunk 1]
```diff
@@ -893,10 +900,32 @@ func (w *worker) commitTransactions(env *environment, txs *transactionsByPriceAn
 	}
 	var coalescedLogs []*types.Log
 
+	//append the tx DAG transaction to the block
+	appendTxDAG := func() {
```
Comment (galaio) [inline miner/worker.go:904]: I think this method is better called in `generateWork`, It may be called multiple times in `commitTransactions`.

Comment (andyzhang2023) [reply miner/worker.go]: Indeed.

File: miner/worker.go [hunk 2]
```diff
@@ -1167,6 +1246,8 @@ func (w *worker) fillTransactions(interrupt *atomic.Int32, env *environment) err
 			return err
 		}
 	}
+	// append a DAG tx at the end of the block
+	w.appendTxDAG(env)
```
Comment (galaio) [inline miner/worker.go:1250]: If there has MEV bundles, this tx may not the last one?

**Review by galaio — APPROVED** [review]

File: miner/worker.go (hunk 2 above)
Comment (andyzhang2023) [reply miner/worker.go]: Yes, indeed.

**Pull request merged**